lib/srv/db/common/kerberos/kinit/kinit.go

```diff
@@ -58,141 +101,59 @@ const (
   pkinit_eku_checking = kpServerAuth
   pkinit_kdc_hostname = {{ .KDCHostName }}
  }`
-	// certTTL is the certificate time to live; 1 hour
-	certTTL = time.Minute * 60
-)
-
-// Provider is a kinit provider capable of producing a credentials cacheData for kerberos
-type Provider interface {
-	// UseOrCreateCredentials uses or updates an existing cacheData or creates a new one
-	UseOrCreateCredentials(ctx context.Context) (cache *credentials.CCache, conf *config.Config, err error)
-}
-
-// PKInit is a structure used for initializing a kerberos context
-type PKInit struct {
-	provider Provider
-}
-
-// UseOrCreateCredentialsCache uses or creates a credentials cacheData.
-func (k *PKInit) UseOrCreateCredentialsCache(ctx context.Context) (*credentials.CCache, *config.Config, error) {
-	return k.provider.UseOrCreateCredentials(ctx)
-}
-
-// New returns a new PKInit initializer
-func New(provider Provider) *PKInit {
-	return &PKInit{provider: provider}
-}
-
-// CommandConfig is used to configure a kinit binary execution
-type CommandConfig struct {
-	// AuthClient is a subset of the auth interface
-	AuthClient windows.AuthInterface
-	// User is the username of the database/AD user
-	User string
-	// Realm is the domain name
-	Realm string
-	// KDCHost is the key distribution center hostname (usually AD server)
-	KDCHost string
-	// AdminServer is the administration server hostname (usually AD server)
-	AdminServer string
-	// DataDir is the Teleport Data Directory
-	DataDir string
-	// LDAPCAPEM contains the same certificate as LDAPCA but in PEM format. It
-	// can be used to embed the LDAPCA into files without needing to convert
-	// it.
-	LDAPCAPEM string
-	// Command is a command generator that generates an executable command
-	Command CommandGenerator
-	// CertGetter is a Teleport Certificate getter that prepares an x509 certificate
-	// for use with windows AD
-	CertGetter CertGetter
-}
-
-// NewCommandLineInitializer returns a new command line initializer using a preinstalled `kinit` binary
-func NewCommandLineInitializer(config CommandConfig) *CommandLineInitializer {
-	cmd := &CommandLineInitializer{
-		auth:               config.AuthClient,
-		userName:           config.User,
-		cacheName:          fmt.Sprintf("%s@%s", config.User, config.Realm),
-		RealmName:          config.Realm,
-		KDCHostName:        config.KDCHost,
-		AdminServerName:    config.AdminServer,
-		dataDir:            config.DataDir,
-		certPath:           fmt.Sprintf("%s.pem", config.User),
-		keyPath:            fmt.Sprintf("%s-key.pem", config.User),
-		binary:             kinitBinary,
-		command:            config.Command,
-		certGetter:         config.CertGetter,
-		ldapCertificatePEM: config.LDAPCAPEM,
-		logger:             slog.Default(),
+	tpl, err := template.New("krb_conf").Parse(krb5ConfigTemplate)
```

Comment:
Should we keep the constant template and the parsed template at the package level to perform the parsing once?

```go
var krb5Template = template.Must(template.New("krb_conf").Parse(krb5ConfigTemplate))
```

Reply:
I considered that, but the parsing isn't that expensive and I'd rather avoid `panic()` (implied by `.Must`) if I can help it. Here we perform some error checking anyway so there is a way to report such error.